tools/lib/graph.py

```diff
@@ -18,13 +18,17 @@ def __init__(self, tuples: EdgeSet) -> None:
             self.nodes.add(parent)
             self.nodes.add(child)
 
-    def copy(self) -> Graph:
+    def copy(self) -> Any:
```

Comment:
Circular type definition in classes are still not possible in mypy.

Reply:
This should work fine if you write `'Graph'`, as a string literal -- see discussion of "forward references" in the mypy docs: http://mypy.readthedocs.io/en/latest/kinds_of_types.html

That is key, because mutually-referring types are common in Python, and convenient, and turning them into `Any` would be sad. :wink: